src/fields/addCounter.jsx

```diff
@@ -0,0 +1,82 @@
+import { Component } from 'react';
+
+export default function addCounter(TextInput) {
```

Comment:
Rather than "textinput" I would call this `ContentComponent` or something. Your HOC does not rely on this argument being specifically a "Text Input", it would work for any component.